src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,231 @@
+variable (in this case, size in bytes). The area of each rectangle is
+proportional to the size the node occupies in the compiled application. Inside
+of each rectangle (call one A), there are additional rectangles that exist one
+level deeper in the data hierarchy (children of A).
+
+To drill into a cell in the treemap, select the cell. This re-roots the tree so
+that the selected cell becomes the visual root of the treemap.
+
+To navigate back, or up a level, use the breadcrumb navigator at the top of the
+treemap.
+
+![Screenshot of treemap breadcrumb navigator]({% asset tools/devtools/treemap_breadcrumbs.png @path %}){:width="100%"}
+
+### Dominator tree and call graph
+
+This section of the page shows code size attribution data (i.e. why a piece of
+code is included in your compiled application). This data is visible
+in the form of a dominator tree as well as a call graph.
+
+#### Using the dominator tree
+
+A [dominator tree](https://en.wikipedia.org/wiki/Dominator_(graph_theory)) is a
+tree where each node's children are those nodes it immediately dominates. A node
+`a` is said to "dominate" a node `b` if every entry point to `b` must go through
+`a`.
+
+To put it in context of app size analysis, imagine `package:a` imports both
+`package:b` and `package:c`, and both `package:b` and `package:c` import
+`package:d`.
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+In this example, `package:a` dominates `package:d`, so the dominator tree for
+this data would look like:
+
+```
+package:a
+|__ package:b
+|__ package:c
+|__ package:d
+```
+
+This information is helpful for understanding why certain pieces of code are
+present in your compiled application. For example, if you are analyzing your
+app size and find an unexpected package included in your compiled app, you can
+use the dominator tree to trace the package to its root source.
+
+![Screenshot of code size dominator tree]({% asset tools/devtools/code_size_dominator_tree.png @path %}){:width="100%"}
+
+#### Using the call graph
+
+A call graph provides similar information to the dominator tree in regards to
+helping you understand why code exists in a compiled application. However,
+instead of showing the one-to-many dominant relationships between nodes of code
+size data like the dominator tree, the call graph shows the many-to-many
+relationships that existing between nodes of code size data.
+
+We'll again use this example:
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+The call graph for this data would link `package:d` to its direct callers,
+`package:b` and `package:c`, instead of its "dominator", `package:a`.
+
+```
+package:a --> package:b -->
+                              package:d
+package:a --> package:c -->
+```
+
+This information is useful for understanding the fine-grained dependencies of
+between pieces of your code (packages, libraries, classes, functions).
```

Comment:
I think you're partially alluding to it. Optional: would be nice to add a small section on why you'd use the dominator tree vs call graph.

Reply:
added info ptal